solidity/contracts/vault/TBTCVault.sol

```diff
@@ -129,7 +130,9 @@ contract TBTCVault is IVault, Ownable {
     ) external override onlyBank {
         require(depositors.length != 0, "No depositors specified");
         for (uint256 i = 0; i < depositors.length; i++) {
-            _mint(depositors[i], depositedAmounts[i]);
+            address depositor = depositors[i];
+            uint256 amount = depositedAmounts[i];
+            _mint(depositor, repayOptimisticMintDebt(depositor, amount));
```

Comment:
In case `repayOptimisticMintDebt(...)` returns 0, do we still want to call `_mint(...)` function? I'm not 100% sure, but it looks like we can mint 0 tokens which consume some gas. Maybe we can skip the minting when there's nothing to mint?

Reply:
I have mixed feelings about it. Our `ERC20WithPermit` implementation [allows to mint 0 tokens](https://github.com/thesis/solidity-contracts/blob/79f658f544232876f70da4bb4b15f365bad6372b/contracts/token/ERC20WithPermit.sol#L158-L170). Similarly, [OpenZeppelin ERC20](https://github.com/OpenZeppelin/openzeppelin-contracts/blob/3d7a93876a2e5e1d7fe29b5a0e96e222afdc4cfa/contracts/token/ERC20/ERC20.sol#L259-L272) implementation. Here is OpenZeppelin thread about it: https://github.com/OpenZeppelin/openzeppelin-contracts/issues/513

Minting 0 will cost some gas but this is quite a rare case and checking if the value is non-zero will also cost some gas while this check will return false in the vast majority of cases. Also, having `Minted` event with 0 value could help in debugging the transaction/building graph.